underactuated.html

```diff
@@ -828,6 +828,23 @@ <h1>Honda's ASIMO vs. passive dynamic walkers</h1>
 
     </exercise>
 
+    <exercise><h1>Trajectory Tracking for Fully-Actuated Systems</h1>
+
+      Take a robot whose dynamics are given by the manipulator equations $${\bf M}(\bq)\ddot\bq +
+      \bC(\bq,\dot\bq)\dot\bq = \btau_g(\bq) + {\bf B}(\bq)\bu.$$  The robot starts in a given initial configuration $\bq(0) = \bq_0$, with a given initial velocity $\dot \bq(0) = \dot \bq_0$.  Suppose the system is fully actuated in all states, i.e., $\textrm{rank} [{\bf B}(\bq)] = \dim[\bq]$ for all $\bq$.  Write, with a brief explanation, whether each of the following statements is true for any twice-differentiable desired trajectory $\bq_{\text{des}} (t)$.
```

Comment:
I LOVE this question!

Reply:
but you do need to say q_des(0) = q0 + initial velocity.

Reply:
can we follow it up with an additional question about the first order equivalent system?  E.g. "we can write any second order equation in the form xdot = f(x,u), where x=[q,qdot].  Given x(0) = [q0, qdot0], can we follow arbitrary trajectories x(t)?".  Alternatively, (or additionally), we could draw an empty phase portrait diagram, and ask "if you answered no, then draw a twice-differentiable trajectory that the system \ddot{q} = u cannot follow".

Reply:
> but you do need to say q_des(0) = q0 + initial velocity.

Why that? This is a question from previous years, the idea here is that even if you can achieve any acceleration you'll never be able to teleopt the system from q(0) to q_des(0), or from qdot(0) to qdot_des(0).

Did you see this more like "I give you a desired q, a desired q_dot, and a desired q_ddot. Can I track these even if one is not the derivative of the others?" I like this too, I can make another question like that!

P.S.: I had a look at the feedback linearization vs underactuation question. I need to study a little more, but I think it could give a better generalization of the def. of underactuation for non 2nd order systems.